### PR TITLE
Fix unsupported type error in Parquet reader

### PR DESCRIPTION
**Pull request by skrzypo987**
## Description

The bug has been introduced in 403, so no release notes are required and the bug will not end up in any version

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java [hunk 1]
```diff
@@ -108,7 +108,6 @@ public static ColumnReader create(PrimitiveField field, DateTimeZone timeZone, b
                 if (isIntegerAnnotation(annotation)) {
```
Comment (raunaqmorarka) [inline lib/trino-parquet/src/main/java/io/trino/parquet/reader/ColumnReaderFactory.java:108]: Let's handle it explicitly here as
```
if (type instanceof TimeType && annotation instanceof TimeLogicalTypeAnnotation timeAnnotation && timeAnnotation.getUnit() == MILLIS) {
        return new FlatColumnReader<>(field, ValueDecoders::getIntToLongDecoder, LONG_ADAPTER);
}
```
and a test in `TestingColumnReader`

**Comment by findepi** [comment]
Can this be tested?

> The bug has been introduced in 403, so no release notes are required and the bug will not end up in any version


Did you mean to mark this PR as `release-blocker`, i.e. required to land before a release goes out?

**Comment by findepi** [comment]
btw 403 is out already

**Comment by skrzypo987** [comment]
> btw 403 is out already

...details

>  Did you mean to mark this PR as release-blocker, i.e. required to land before a release goes out?

No. After consulting things with Raunaq, we came to a conclusion that this is not a bug at all, because the previous behavior was not correct. So we exchanged incorrect read for fail-fast exception, which is good.

I will be posting this feature today anyway, this time working properly with proper testing, so there is good change it will land in 40-next

**Comment by skrzypo987** [comment]
Closing in favour of https://github.com/trinodb/trino/pull/15200

**Pull request closed without merging**